### ISSUE TITLE
JSON field ignored during load when it's an array

### ISSUE THREAD

**Issue by fubar-coder** [dotnetcore/FreeSql#653]
Test case:

```csharp
using var freeSql = new FreeSql.FreeSqlBuilder()
	.UseConnectionString(DataType.Sqlite, "Data Source=C:/Spielwiese/test.sqlite")
	.UseAutoSyncStructure(true)
	.Build();

freeSql.UseJsonMap();

freeSql.CodeFirst.ConfigEntity<TestClass>(cf =>
{
	cf.Property(p => p.Name).IsNullable(false);
	cf.Property(p => p.Tags).JsonMap();
});

freeSql.Insert<TestClass>(
	new TestClass("test 1")
	{
		Tags = new[] { "a", "b" },
	})
	.ExecuteAffrows();
var records = freeSql.Queryable<TestClass>().ToList();
records.Dump();

record TestClass(string Name)
{
	public Guid Id { get; set; }
	public string[] Tags { get; init; } = Array.Empty<string>();
}
```

The `Tags` property of the loaded records is an empty record, but the data is stored correctly. I guess that the ORM sees the `Tags` property as a navigation property and therefore skips the JSON deserialization.

Tested with 2.0.105. The 2.3 preview throws an exception that it's unable to cast from `string[]` to `string`.

**Comment by 2881099** [comment]
Modify Test Case:

```c#
record TestClass(string Name)
{
    public Guid Id { get; set; }
    public TestClassTagInfo Tags { get; init; }
}
record TestClassTagInfo //Important changes
{
    public string[] Tags { get; init; } = Array.Empty<string>();
}

    using (var freeSql = new FreeSql.FreeSqlBuilder()
        .UseConnectionString(DataType.Sqlite, "Data Source=test11.sqlite")
        .UseAutoSyncStructure(true)
        .UseNoneCommandParameter(true)
        .UseMonitorCommand(cmd => Trace.WriteLine(cmd.CommandText))
        .Build())
    {

        freeSql.UseJsonMap();

        freeSql.CodeFirst.ConfigEntity<TestClass>(cf =>
        {
            cf.Property(p => p.Name).IsNullable(false);
            cf.Property(p => p.Tags).JsonMap();
        });

        freeSql.Insert<TestClass>(
            new TestClass("test 1")
            {
                Tags = new TestClassTagInfo { Tags = new[] { "a", "b" } },
            })
            .ExecuteAffrows();
        var records = freeSql.Queryable<TestClass>().ToList();

    }
```

The string [] type is not supported in FreeSQL JsonMap.

It is recommended to use the new type, which can be better supported.

At present, there are some limitations in the use of jsonmap, and we will improve it later.


thank.

**Comment by fubar-coder** [comment]
@luoyunchong This doesn't seem to be a viable solution, because the resulting JSON is different

**Comment by 2881099** [comment]
Problem solved, v3.2.666-preview20220809
